src/helpers/prss.rs

```diff
@@ -94,14 +94,68 @@ impl PrssSpace {
         let (l, r): (F, F) = self.generate_fields(index);
         l + r
     }
+
+    /// Turn this space into a pair of random number generators, one that is shared
+    /// with the left and one that is shared with the right.
+    /// Nothing prevents this from being used after calls to generate values,
+    /// but it should not be used in both ways.
+    #[must_use]
+    pub fn as_rngs(&self) -> (PrssRng, PrssRng) {
```

Comment:
would it be better if this method sets the counter to u32::MAX and also consumes self?

Reply:
Take a look at the first commit.

I would love to have this consume self, but that is not compatible with the way in which this object is used throughout.